test/test_util.py

```diff
@@ -138,34 +139,34 @@ def test_parse_url_normalization(self):
             actual_normalized_url = parse_url(url).url
             self.assertEqual(actual_normalized_url, expected_normalized_url)
 
-    parse_url_host_map = {
-        'http://google.com/mail': Url('http', host='google.com', path='/mail'),
-        'http://google.com/mail/': Url('http', host='google.com', path='/mail/'),
-        'http://google.com/mail': Url('http', host='google.com', path='mail'),
-        'google.com/mail': Url(host='google.com', path='/mail'),
-        'http://google.com/': Url('http', host='google.com', path='/'),
-        'http://google.com': Url('http', host='google.com'),
-        'http://google.com?foo': Url('http', host='google.com', path='', query='foo'),
+    parse_url_host_map = [
+        ('http://google.com/mail', Url('http', host='google.com', path='/mail')),
+        ('http://google.com/mail/', Url('http', host='google.com', path='/mail/')),
+        ('http://google.com/mail', Url('http', host='google.com', path='mail')),
+        ('google.com/mail', Url(host='google.com', path='/mail')),
+        ('http://google.com/', Url('http', host='google.com', path='/')),
+        ('http://google.com', Url('http', host='google.com')),
+        ('http://google.com?foo', Url('http', host='google.com', path='', query='foo')),
 
         # Path/query/fragment
-        '': Url(),
-        '/': Url(path='/'),
-        '#?/!google.com/?foo#bar': Url(path='', fragment='?/!google.com/?foo#bar'),
-        '/foo': Url(path='/foo'),
-        '/foo?bar=baz': Url(path='/foo', query='bar=baz'),
-        '/foo?bar=baz#banana?apple/orange': Url(path='/foo', query='bar=baz', fragment='banana?apple/orange'),
+        ('', Url()),
+        ('/', Url(path='/')),
+        ('#?/!google.com/?foo#bar', Url(path='', fragment='?/!google.com/?foo#bar')),
+        ('/foo', Url(path='/foo')),
+        ('/foo?bar=baz', Url(path='/foo', query='bar=baz')),
+        ('/foo?bar=baz#banana?apple/orange', Url(path='/foo', query='bar=baz', fragment='banana?apple/orange')),
 
         # Port
-        'http://google.com/': Url('http', host='google.com', path='/'),
-        'http://google.com:80/': Url('http', host='google.com', port=80, path='/'),
-        'http://google.com:80': Url('http', host='google.com', port=80),
+        ('http://google.com/', Url('http', host='google.com', path='/')),
+        ('http://google.com:80/', Url('http', host='google.com', port=80, path='/')),
+        ('http://google.com:80', Url('http', host='google.com', port=80)),
 
         # Auth
-        'http://foo:bar@localhost/': Url('http', auth='foo:bar', host='localhost', path='/'),
-        'http://foo@localhost/': Url('http', auth='foo', host='localhost', path='/'),
-        'http://foo:bar@baz@localhost/': Url('http', auth='foo:bar@baz', host='localhost', path='/'),
-        'http://@': Url('http', host=None, auth='')
-    }
```

Comment:
There's two keys with different values for `http://google.com/mail` and `http://google.com/` and that will silently drop one of the test cases.